spec/controllers/samples_controller_spec.rb

```diff
@@ -11,16 +11,46 @@
 
     describe "GET raw_results_folder (nonadmin)" do
       it "can see raw results on the user's own samples" do
+        project = create(:project, users: [@joe])
+        sample = create(:sample, project: project, user: @joe)
+        get :raw_results_folder, params: { id: sample.id }
+        expect(response).to have_http_status :success
+      end
+
+      it "cannot see raw results on another user's sample" do
         project = create(:project, users: [@joe])
         sample = create(:sample, project: project)
         get :raw_results_folder, params: { id: sample.id }
         expect(response).to have_http_status :unauthorized
       end
+    end
 
-      it "cannot see raw results on another user's sample" do
+    describe "GET results_folder (nonadmin)" do
+      it "can see raw results on the user's own samples" do
         project = create(:project, users: [@joe])
         sample = create(:sample, project: project, user: @joe)
-        get :raw_results_folder, params: { id: sample.id }
+        get :results_folder, params: { id: sample.id }
```

Comment:
It seems like there is a mismatch here between "can see raw results" and the endpoint that's actually tested (results_folder, not raw_results_folder)